### ISSUE TITLE
GitDiffReporter._parse_lines calculate (added_lines, deleted_lines) may be unexpected?

### ISSUE THREAD

**Issue by yujunjiex** [Bachmann1234/diff_cover#175]
func of _parse_hunk() explain hunk line is :

> @@ -k,l +n,m @@ TEXT
> 
> where `k,l` represent the start line and length before the changes
> and `n,m` represent the start line and length after the changes.

and only returned 'n' in the code

https://github.com/Bachmann1234/diff_cover/blob/01f91760321cee1ad28cfa0d801c4acd8b9765a6/diff_cover/diff_reporter.py#L420-L435

func of _parse_lines() call _parse_hunk() to get 'n', and seem think that both variables of 'current_line_new' and 'current_line_old' are equal to 'n'

https://github.com/Bachmann1234/diff_cover/blob/01f91760321cee1ad28cfa0d801c4acd8b9765a6/diff_cover/diff_reporter.py#L346-L351

In my opinion, more detailed would be :
> where `k,l` represent the start line and length before the changes when there have rows '-'
> and `n,m` represent the start line and length after the changes when there have rows '+'

There are the following example:

**example1:**
```
--- a/demo.go
+++ /dev/null
@@ -1,1 +0,0 @@
-package main
```

if use '0' as begin of line, 'deleted_lines' will contain [0].

**example2:**
```
--- a/demo2.go
+++ b/demo2.go
@@ -5,2 +4,0 @@ import (
-       "context"
-       "fmt"
@@ -7,0 +6 @@ import (
+       "os"
```

if use '4' as begin of line, 'deleted_lines' will contain [4, 5], but [5, 6] is more consistent with the definition of 'deleted_lines'.

So, should 'current_line_new' and 'current_line_old' be calculated separately by n and k?

**Comment by kasium** [comment]
I'm not quite sure with this since I'm new to the project, but let me try to give an answer.
As fas as I can see and my tests have shown, diff-cover works only on the "new" file, so it make sense to only get the "n". What should be the purpose to have the information, that the deleted line was e.g. 2?

Coverage reports are only using the current files (so in git diff the new). Therefore we only need to consider the new line numbers.

@Bachmann1234  maybe you can add here something, but I guess it's a small understanding issue how diff-cover works with deleted line numbers

**Comment by Bachmann1234** [comment]
Sadly I wont be too much help here as this logic was written before I took over the project and I never really learned these details all that well

**Comment by kasium** [comment]
Let me try to dig a little bit deeper into it

**Comment by kasium** [comment]
So I took some more time to understand the code here. I guess the naming is a little bit misleading here.
In my understanding, the deleted lines are only needed to keep track of differences between staged, unstaged and untracked files. If diff-cover would e.g. only support staged files we don't need to parse the deleted lines at all.
With that being said, the deleted lines "definition" is designed to be used internally in diff-cover and not be somehow compliant to the git-definition. Therefore, the lines are using the actual status of the file and not the both the old and new status.

Basically, I can confirm that this makes totally sense inside diff-cover but of course not outside of it. The parser should only be used in this project therefore.

@Bachmann1234 since the deleted line calculation works fine and I see this only as a misunderstanding of the internal machinery of diff-cover you might want to close this issue
